### DISCUSSION TITLE
Enhancement: Blind Accessibility

### DISCUSSION THREAD

**Discussion by felix-nuh** [FunkinCrew/Funkin#1633] [Ideas]
# Blind Accessibility 

Blind gamers are often shafted when it comes to popular video games. Many are aware of their friends playing exciting new titles that offer no accessibility to them. Many games could use simple design changes to make the experience more usable to blind people just from a basic analysis.

# How would it be implemented?

First, I suggest also talking to people on both [r/blind](https://www.reddit.com/r/Blind/) and [audiogames](https://audiogames.net), two major sites for blind users, to experiment with the game. This way, you can test your implementation with blind users who can inform you of important things you miss or user experience.

I am sighted, but my suggestions would be the following;
- SAPI support as a bare minimum - NVDA support as well if you can manage it. If you want to VO all of your menus, descriptions and scripts, this is also an option.
- Focus to menus, and reading focused menu option as it's selected.
- Descriptive text of HUDs, menus, ect. available from additional buttons. This really adds to immersion. (Possibly allow people to "view" their environment on a pause menu?)
- important HUD elements must be readable through a TTS through a simple key or button press.
- Sound alert cues - when a player is expected to press a button, a specific cue for that button should be played.
- Keyboard binding. 

I haven't played the game yet so if something is already implemented, just ignore it. I've seen video online of how the game is played. Obviously more suggestions are open. Also, engage back and forth with these communities to produce a more usable accessible game, it really has potential.

Thank you for reading. I don't normally use github by the way so I will not be monitoring this issue directly but I wanted the developers to be aware of it.

**Comment by oketariq** [comment]
I really like the idea of support for people who are blind, but do not believe that this should be a feature. A key part of Friday Night Funkin' is sight, to hit notes at the exact moments you need to according to the song. That's what games like these are about. There is also a lot of loud music in the game that may disturb the beeps and make them unhearable. Overall i think it's a great idea, just not suited for this game. This is my opinion, by the way.

**Comment by felix-nuh** [comment]
Hello,

I disagree entirely with your analysis. I specialize in blind accessibility
in game development, have had over 7 years of experience, and I hope that
these suggestions can help better influence your decision.

> A key part of Friday Night Funkin' is sight, to hit notes at the exact
moments you need to according to the song. That's what games like these are
about

Actually, Rhythm games are based simply on pushing inputs at specific
times. Signifying when that sound should be pressed with carefully timed
beeps offers nearly the same functionality as a visual cue. This feature
can be disabled normally and be activated by an option setting. The sounds
can be cued to play when the trigger animation plays so that a player knows
what button to press.

In fact, Rhythm games are not particularly hard to make accessible in
general.

> There is also a lot of loud music in the game that may disturb the beeps
and make them unhearable

You should include a basic volume slider for music and sound effects in
your options menu. This will also improve your game's presentation overall.

All of this is very simple to build and should not require more than a few
days worth of development.

**Comment by T0paz** [comment]
Adding to this discussion, most minigames in Rhythm Heaven can be played without looking at the screen because of their focus on audio design. Most FNF weeks could be modified to provide a "blind" variant, with a focus on hitting the correct timings rather than specific directions, as long as there are vocals matching the corresponding difficulty and unique inputs are either removed or reworked. Technically, you can tell which buttons to press by the vocals already, but this can sometimes be inconsistent, requiring a rework of how vocals translate (not easy).

**Comment by felix-nuh** [comment]
I like this idea a lot, it can provide an additional gameplay mode as well
to other players.

I was listening to Parappa the Rapper and I noticed that as well,
repetition of input phrases (groups of commands correlated with music)
really helps direct inputs as well. Improving the audio design will greatly
improve presentation overall as well.

Don't forget about key rebinding as well. You can build a rebind manager
that contains all the inputs as variables, and the input functions can
check those instead of static keys. While an in-game interface is
preferred, using an external text file is also acceptable.
